pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -148,3 +167,143 @@ func (f *Factory) importOVAIfMissing(ctx context.Context, templateName, ovaURL s
 
 	return nil
 }
+
+func (f *Factory) resizeDisk(ctx context.Context, templateFullPath, templateName, datacenter string) error {
```

Comment:
From the above comment, lets change this to just have a function to validate and not resize user imported templates. For the validation function too, lets try and have one function which we can re-use.